connect/file/src/main/java/org/apache/kafka/connect/file/FileStreamSourceConnector.java

```diff
@@ -101,4 +105,49 @@ public ExactlyOnceSupport exactlyOnceSupport(Map<String, String> props) {
                 : ExactlyOnceSupport.UNSUPPORTED;
     }
 
+    @Override
+    public boolean alterOffsets(Map<String, String> connectorConfig, Map<Map<String, ?>, Map<String, ?>> offsets) {
+        AbstractConfig config = new AbstractConfig(CONFIG_DEF, connectorConfig);
+        String filename = config.getString(FILE_CONFIG);
+        if (filename == null || filename.isEmpty()) {
+            throw new ConnectException("Offsets cannot be modified if the '" + FILE_CONFIG + "' configuration is unspecified. " +
+                    "This is because stdin is used for input and offsets are not tracked.");
+        }
```

Comment:
Hah this seems very similar to what we've been discussing on the MM2 PR and has somewhat comparable flexibility - user friendliness tradeoffs 😅

Allowing users to modify offsets when the connector is currently configured to read from stdin could be useful in order to clean up stale offsets. However, it could be slightly misleading to accept offsets modification requests and return a successful response and message when in actuality the operation was a no-op as far as the connector's task is concerned. I'd also like to note for posterity that it isn't possible to detect past connector configurations or previous offsets in the `SourceConnector::alterOffsets` method. Both the following cases seem uncommon enough that either approach should be acceptable - 

- An unfamiliar user attempts to modify offsets for a file stream source connector configured to read from stdin and expects the connector to re-produce past data.
- A user configures a file stream source connector to read from a file, then reconfigures it to read from stdin, and then wants to clean up previous offsets.

Reply:
Fair enough--I think it's fine to leave as-is for now 👍

I'll merge, but depending on the outcome of the discussion around loosening partition validation for tombstones in https://github.com/apache/kafka/pull/14005, we may want to tweak this method in a follow-up to do the same.